### DISCUSSION TITLE
Can docstrings be displayed on hover in addition to type signature?

### DISCUSSION THREAD

**Discussion by pmalacho-mit** [erictraut/pyright-playground#2] [Q&A] [answered]
Hello! I was wondering if, in a browser environment, it'd be possible to configure pyright to display docs in addition to type information on hover (similar to the experience in vscode, I'll paste some comparison screenshots)? From some googling around, it seems like doc information might be handled by pylance instead of pyright, is that correct? Something to confuse me slightly more is that using [vscode.dev](https://vscode.dev/) with the python extension enabled also does **NOT** show docs. Perhaps this is one of the features that is limited in the browser? 

Pyright Playground
<img width="316" alt="Screen Shot 2024-01-31 at 10 24 31 AM" src="https://github.com/erictraut/pyright-playground/assets/95306112/57939f3b-af79-4e0f-9f02-40bb758e644e">

Vscode
<img width="583" alt="Screen Shot 2024-01-31 at 10 26 19 AM" src="https://github.com/erictraut/pyright-playground/assets/95306112/579c8623-a8bb-4f30-87f2-846f95fde191">

Vscode.dev

<img width="350" alt="Screen Shot 2024-01-31 at 10 25 59 AM" src="https://github.com/erictraut/pyright-playground/assets/95306112/8d6f53d7-61f9-4884-a2c1-0cb954b4019c">

**Comment by erictraut** [comment, marked as answer]
The hover functionality in pyright playground simply calls through to the pyright LSP and displays what it returns. This does include docstrings.

The reason you're not seeing any docstring for `str.split` is that the type information for this method comes from the typeshed stub `builtins.pyi`, and the stub doesn't include any docstrings. 

The "Vscode" screen shot was presumably using Pylance, not pyright. Pylance is Microsoft's premium language server that builds on pyright's functionality. It bundles additional information for stdlib modules including these docstrings. The base pyright functionality doesn't include this. If you're building on top of pyright, you'd need to similarly add this functionality.

**Reply by pmalacho-mit** [reply]
Ah, thank you for the clarification, @erictraut ! Something that is confusing to me (which is likely out of scope for this question, apologies) is why on `vscode.dev` with the official ms python extension in use, which (I believe) uses pylance under the hood, I similarly don't see docs. Do you know why that might be? I just want to make sure there's not a browser specific reason that pursuing docstrings is a bad idea (perhaps simply their size?).

From your answer, it sounds like if the stubs included docstrings then pyright would display them -- which is great! I'm using typeshed's stubs, which do not include docs, which seems like at least some topic of conversation: https://github.com/python/typeshed/issues/4881

Thanks for the help!

[EDIT]

Oh, too funny -- I see in the [linked thread](https://mail.python.org/archives/list/python-ideas@python.org/thread/QKJWPZ6XULFWNRFZYNTXXDC26UVQPFNG/#QKJWPZ6XULFWNRFZYNTXXDC26UVQPFNG) to the github issue I referenced you are actually a contributor, @erictraut . Your reply there is also helpful for me understanding this matter, thanks!

**Reply by erictraut** [reply]
> Do you know why that might be?

You'd need to ask someone on the Pylance team for a definitive answer, but I suspect it's because Pylance running in a browser doesn't have access to the stdlib docstring information bundled with the Pylance extension. It's presumably quite large and would be too costly to download to the browser (client) every time someone uses vscode.dev.

> it sounds like if the stubs included docstrings then pyright would display them

That's correct. You can see docstrings appear in hover text if you create a function with a docstring in the [pyright playground](https://pyright-play.net/?reportMissingTypeArgument=true&code=CYUwZgBGCuB2DGAKAlALgLACgI4gIgIgEEJgB7eAZwBcAnAS1gHMIQAPAQwFsAHAGxD4CWLAGIIACTIA3ELQgy5%2BGAjwQARiD5kA7hGpkIlEIOoALQeSp1GTAHRYVSZFiA) and then hover over a reference to that function.

**Reply by pmalacho-mit** [reply]
Great, thank you, @erictraut !
